src/main/scala/piconot/RightHand.scala

```diff
@@ -0,0 +1,24 @@
+package piconot
+
+object RightHand extends PicoBot {
+
+  use map "resources/maze.txt"
+
+  val faceN, faceE, faceW, faceS = States
+
+  faceN-
+  -N -> N and faceE
+  +N -> faceW
+
+  faceE-
+  -E -> E and faceS
+  +E -> faceN
+
+  faceS-
+  -S -> S and faceW
+  +S -> faceE
+
+  faceW-
+  -W -> W and faceN
+  +W -> faceS
+}
```

Comment:
Though the 8 rule solution to the maze gets you extra credit in CS5, I would argue that the 12 rule solution (or even the 16 rule one!) is much more easily understandable than the 8 rule one, and if a goal of this language is better readability I would recommend providing the simpler RightHand algorithm as an example program for a new user.